scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -1667,3 +1670,459 @@ cdef FibonacciNode* remove_min(FibonacciHeap* heap) noexcept:
 #        print_node(heap.min_node)
 #    else:
 #        print("[empty heap]")
+
+######################################################################
+
+# Author: Tomer Sery  -- <tomersery28@gmail.com>
+# License: BSD, (C) 2024
+
+def yen(
+    csgraph,
+    source,
+    sink,
+    K,
+    *,
+    directed=True,
+    return_predecessors=False,
+    unweighted=False,
+):
+    """
+    yen(csgraph, source, sink, K, *, directed=True, return_predecessors=False,
+        unweighted=False)
+
+    Yen's K-Shortest Paths algorithm on a directed or undirected graph.
+
+    .. versionadded:: 1.13.0
+
+    Parameters
+    ----------
+    csgraph : array or sparse array, 2 dimensions
+        The N x N array of distances representing the input graph.
+    source : int
+        The index of starting node for the paths.
+    sink : int
+        The index of the ending node for the paths.
+    K : int
+        The number of shortest paths to find.
+    directed : bool, optional
+        If True (default), then find the shortest path on a directed graph:
+        only move from point i to point j along paths ``csgraph[i, j]``.
+        If False, then find the shortest path on an undirected graph: the
+        algorithm can progress from point i to j along ``csgraph[i, j]`` or
+        ``csgraph[j, i]``
+    return_predecessors : bool, optional
+        If True, return the size (N, N) predecessor matrix.
+    unweighted : bool, optional
+        If True, then find unweighted distances. That is, rather than finding
+        the path between each point such that the sum of weights is minimized,
+        find the path such that the number of edges is minimized.
+
+    Returns
+    -------
+    dist_array : ndarray
+        The M array of shortest distances between the source and sink nodes.
```

Comment:
Is this 'array of size `M`'? I think that would be clearer. Some more backticks would be good below too.

Reply:
Done